core/src/main/java/io/questdb/cairo/BitmapIndexWriter.java

```diff
@@ -48,17 +49,27 @@
     private final BitmapIndexUtils.ValueBlockSeeker SEEKER = this::seek;
 
     public BitmapIndexWriter(CairoConfiguration configuration, Path path, CharSequence name) {
-        of(configuration, path, name);
+        of(
+                configuration,
+                path,
+                name,
+                configuration.getDataIndexKeyAppendPageSize(),
+                configuration.getDataIndexValueAppendPageSize()
+        );
+    }
+
+    public BitmapIndexWriter(CairoConfiguration configuration, Path path, CharSequence name, long keyAppendPageSize, long valueAppendPageSize) {
+        of(configuration, path, name, keyAppendPageSize, valueAppendPageSize);
     }
 
     public BitmapIndexWriter() {
     }
 
-    public static void initKeyMemory(VirtualMemory keyMem, int blockValueCount) {
+    public static void initKeyMemory(MemoryMA keyMem, int blockValueCount) {
 
         // block value count must be power of 2
         assert blockValueCount == Numbers.ceilPow2(blockValueCount);
-
+        keyMem.toTop();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;keyMem&#39; (index: 0 | type: MemoryMA). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other validated parameters: &#39;blockValueCount&#39;. The same parameter type is validated here for example: core/src/main/java/io/questdb/cairo/O3OpenColumnJob.java:852. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness (for example, using the @NonNull annotation as [described in the Lombok library](https://projectlombok.org/features/NonNull)), emptiness, or equality. Or to prevent direct calls to it, [reduce the method&#39;s visibility](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html).

 Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).